### ISSUE TITLE
[2.0.4] Many-to-Many relations with conditions 

### ISSUE THREAD

**Issue by alitalaghat** [phalcon/cphalcon#10609]
After upgrade to 2.0.4, ManyToMany relation don't apply condition.
```
    $this->hasManyToMany('id', 'Core\Models\PostsMetas', 'postId',
            'value', 'Core\Models\Seasons','id', array(
                'alias' => 'seasons',
                'params' => [
                    'conditions' => 'key="season"'
                ]
            ));
```
Example result:
SQL result in 2.0.4 :
```
SELECT `seasons`.`season_id`, `seasons`.`season_slug`, `seasons`.`season_title` FROM `seasons`  INNER JOIN `posts_metas` ON `posts_metas`.`meta_value` = `seasons`.`season_id` WHERE `posts_metas`.`meta_post_id` = '18'
```
SQL result in 2.0.3 :
```
SELECT `seasons`.`season_id`, `seasons`.`season_slug`, `seasons`.`season_title` FROM `seasons`  INNER JOIN `posts_metas` ON `posts_metas`.`meta_value` = `seasons`.`season_id` WHERE (`posts_metas`.`meta_key` = 'season') AND (`posts_metas`.`meta_post_id` = '18')
```

**Comment by andresgutierrez** [comment]
Fixed in the 2.0.x branch

**Comment by sxpd7788** [comment]
```
$this->hasManyToMany(
            'id',
            TagRelation::class,
            'relationid', 'tagid',
            Tag::class,
            'id',
            [
                'params' => [
                    'conditions' => '[' . TagRelation::class . '].type = "1" ',
                ],
                'alias'=>'tag'
            ]
        );
```

I get this error
Phalcon\Mvc\Model\Exception: Syntax error, unexpected token ), near to ' AND ([warmup\models2\TagRelation].[relationid] = :APR0:)', when parsing: SELECT [warmup\models2\Tag].* FROM [warmup\models2\Tag] INNER JOIN [warmup\models2\TagRelation] ON [warmup\models2\TagRelation].[tagid] = [warmup\models2\Tag].[id] WHERE (([warmup\models2\TagRelation].type = "1" ) AND ) AND ([warmup\models2\TagRelation].[relationid] = :APR0:) (276)
